### ISSUE TITLE
node_modules file is not ignored correctly if it contains eslint config

### ISSUE THREAD

**Issue by JounQin** [eslint/eslint#14686]
Related: https://github.com/microsoft/vscode-eslint/issues/1240

And also #14533, #14546. They are both closed, but actually it is a bug IMO.

`node_modules` is already in my `.eslintignore`. (`**/node_modules` does not work too)

![image](https://user-images.githubusercontent.com/8336744/121353565-95129b00-c960-11eb-8b9b-71203113e747.png)

---

I don't know should it be fixed in ESLint core or the related vscode extension. And I don't know if any other related editors share the same issue.

---

**Tell us about your environment**

* **ESLint Version:** v7.27.0
* **Node Version:** v12.22.1
* **npm Version:** 7.18.1
* **Operating System:** macOS 12

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

N/A

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

N/A

```bash
yarn eslint node_modules/stylelint/lib/cli.js
```

**What did you expect to happen?**

report file ignored warning

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

```log
❯ yarn eslint node_modules/stylelint/lib/cli.js                                                                                                                          (master|●11)

yarn run v1.22.10
$ /Users/JounQin/Workspaces/Alauda/icarus/node_modules/.bin/eslint node_modules/stylelint/lib/cli.js

Oops! Something went wrong! :(

ESLint: 7.27.0

ESLint couldn't find the config "stylelint" to extend from. Please check that the name of the config is correct.

The config "stylelint" was referenced from the config file in "/Users/JounQin/Workspaces/Alauda/icarus/node_modules/stylelint/package.json".

If you still have problems, please stop by https://eslint.org/chat/help to chat with the team.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Steps to reproduce this issue:**

https://github.com/alauda/k8s-form-in-action

run `yarn && yarn eslint node_modules/stylelint/lib/cli.js`

**Are you willing to submit a pull request to fix this bug?**

Yes

**Comment by aladdin-add** [comment]
just curious why you want to lint node_modules?

**Comment by JounQin** [comment]
@aladdin-add I'm not, `vscode-eslint` is doing this.

> I don't know should it be fixed in ESLint core or the related vscode extension. And I don't know if any other related editors share the same issue.

**Comment by ljharb** [comment]
It seems like the fix in eslint would be "check eslintignore before attempting to load the eslint config for a given file"?

**Comment by JounQin** [comment]
> It seems like the fix in eslint would be "check eslintignore before attempting to load the eslint config for a given file"?

Yeah, it should be the solution IMO.

**Comment by nzakas** [comment]
@JounQin please fill out the correct [issue template](https://github.com/eslint/eslint/blob/master/.github/ISSUE_TEMPLATE/BUG_REPORT.md). Once we have all of those details fully listed in this issue we can evaluate further. Thanks.

**Comment by JounQin** [comment]
@nzakas Updated.

**Comment by nzakas** [comment]
@JounQin thanks. Someone will take a look soon.

**Comment by snitin315** [comment]
Hi @JounQin, I tried with your reproduction and I was able to reproduce the error, but when I tried it on a different project ESLint did show a warning as per -  https://github.com/eslint/eslint/blob/aaf65e629adb74401092c3ccc9cb4e4bd1c8609b/lib/cli-engine/cli-engine.js#L290

<img width="918" alt="Screenshot 2021-07-10 at 1 16 35 PM" src="https://user-images.githubusercontent.com/46647141/125156158-adedb680-e181-11eb-96cd-62f7c0b0d0c9.png">


When I compared both cases I found that if a `node-module/dependency/pakage.json` has an `eslintConfig` field, ESLint is considering it despite the fact that's it's in `.eslintignore`. 

<img width="1078" alt="Screenshot 2021-07-10 at 1 32 04 PM" src="https://user-images.githubusercontent.com/46647141/125156477-3f115d00-e183-11eb-8bb5-827957d7e5c4.png">


That definitely looks like a bug to me. It would be great if someone else from the team can confirm it too.

**Comment by JounQin** [comment]
> When I compared both cases I found that if a node-module/dependency/pakage.json has an eslintConfig field, ESLint is considering it despite the fact that's it's in .eslintignore.

Right, that is what I described `if it contains eslint config`.

**Comment by snitin315** [comment]
Yeah, are you still interested in sending a fix? otherwise, I will take care of it.

**Comment by JounQin** [comment]
If it is confirmed should be fixed, I'll take a look at tonight or tomorrow.

**Comment by snitin315** [comment]
If someone can confirm from @eslint/eslint-tsc, we can mark it as accepted.

**Comment by JounQin** [comment]
Yeah, I'm still waiting.

**Comment by nzakas** [comment]
Sorry, 2/3 of the TSC have been away leaving me covering a lot. This is still on my list to confirm, hopefully soon.

**Comment by nzakas** [comment]
I've confirmed this is most likely a bug, however, I don't think it's one we want to fix at this point. The problem is the complexity of the way the eslintrc config files work, and any time we touch or change things, we run the risk of introducing other errors along the way. It's this complexity that has led us towards a new, simpler config system that will replace it. We made the decision to freeze eslintrc to minimize any further disruptions to our users while we work on transitioning to a new config system.
